frontend/src/app/modules/matches/matches-computations.js

```diff
@@ -68,3 +77,104 @@ export const computeWinRatio = (playerId, playerMatches) => {
   const wonMatchesCount = playerMatches.filter(match => didPlayerWin(playerId, match)).length
   return (playerMatches.length > 0) ? (wonMatchesCount / playerMatches.length) : 0
 }
+
+export const computeDays = matchChanges => {
+  let worstDayKey = ''
+  let bestDayKey = ''
+  const daysMap = {}
+  for (const match of matchChanges) {
+    const key = match.date.toLocaleDateString()
+    const value = Number(match.ratingChangeString)
+    if (!Object.hasOwnProperty.call(daysMap, key)) {
+      daysMap[key] = {
+        date: key,
+        value,
+      }
+    } else {
+      daysMap[key]['value'] += value
+    }
+
+    bestDayKey = updateMaxFromMap(daysMap[bestDayKey] ?? {}, 'value', daysMap[key]).date
+    worstDayKey = updateMinFromMap(daysMap[worstDayKey] ?? {}, 'value', daysMap[key]).date
+  }
+
+  return {
+    bestDay: daysMap[bestDayKey],
+    worstDay: daysMap[worstDayKey],
+  }
+}
+
+export const computeTeammates = (playerId, playerMatches) => {
+  const teammates = findSignificanPlayers(playerId, playerMatches, getTeamMates)
+  return {
+    mostFrequentTeammate: teammates.mostFrequentPlayer,
+    leastFrequentTeammate: teammates.leastFrequentPlayer,
+    mostSuccessTeammate: teammates.mostSuccessPlayer,
+    leastSuccessTeammate: teammates.leastSuccessPlayer,
+  }
+}
+
+export const computeOpponents = (playerId, playerMatches) => {
+  const opponents = findSignificanPlayers(playerId, playerMatches, getOpponents)
+  return {
+    mostFrequentOpponent: opponents.mostFrequentPlayer,
+    leastFrequentOpponent: opponents.leastFrequentPlayer,
+    mostSuccessOpponent: opponents.mostSuccessPlayer,
+    leastSuccessOpponent: opponents.leastSuccessPlayer,
+  }
+}
+
+const findSignificanPlayers = (playerId, playerMatches, playersProvider) => {
+  let mostFrequentPlayer = {}
+  let leastFrequentPlayer = {}
+  let mostSuccessPlayer = {}
+  let leastSuccessPlayer = {}
+  const playersMap = {}
+  for (const match of playerMatches) {
+    const players = playersProvider(playerId, match)
+    for (const player of players) {
+      if (!Object.hasOwnProperty.call(playersMap, player.id)) {
```

Comment:
Personally, I don't like maps due to the necessity to always call `get` and `set` instead of direct access. Also, it would require some more edits as the function will now return map instead of object. I made the code simpler with the second approach you mentioned [here](https://github.com/salsita/foosball-rating/pull/114/commits/3d398530cdf75f980e6344e885e4f18dd919311a#diff-304dca28d92d292252e86b7e83fc1569)